admin/app/controllers/admin/ReaderRevenueAdminController.scala

```diff
@@ -26,45 +26,61 @@ class ReaderRevenueAdminController(wsClient: WSClient, val controllerComponents:
     NoCache(Ok(views.html.readerRevenue.bannerDeploys(ReaderRevenueRegion.allRegions)))
   }
 
-  def redeployContributionsBanner(strRegion: String): Action[AnyContent] = Action.async { implicit request =>
-    ReaderRevenueRegion.fromString(strRegion).fold(Future(redeployFailed(new Throwable("attempted to redeploy banner in unknown region")))){ region: ReaderRevenueRegion =>
+  def renderSubscriptionsBannerAdmin: Action[AnyContent] = Action { implicit request =>
+    NoCache(Ok(views.html.readerRevenue.subscriptionsBannerDeploys(ReaderRevenueRegion.allRegions)))
+  }
+
+  def redeployContributionsBanner(region: String): Action[AnyContent] = redeployBanner(region, ContributionsBanner)
+
+  def redeploySubscriptionsBanner(region: String): Action[AnyContent] = redeployBanner(region, SubscriptionsBanner)
+
+  def redeployBanner(region: String, bannerType: BannerType): Action[AnyContent] = Action.async { implicit request =>
+    ReaderRevenueRegion.fromString(region).fold(Future(redeployFailed(new Throwable("attempted to redeploy banner in unknown region"), bannerType))){ region: ReaderRevenueRegion =>
       val requester: String = UserIdentity.fromRequest(request) map(_.fullName) getOrElse "unknown user (dev-build?)"
       val time = DateTime.now
-      val jsonLog: JsValue = Json.toJson(ContributionsBannerDeploy(time))
-      val message = s"Contributions banner in ${region.name} redeploy by $requester at ${time.toString}}"
+      val jsonLog: JsValue = Json.toJson(BannerDeploy(time))
+      val message = s"${bannerType.name} banner in ${region.name} redeploy by $requester at ${time.toString}}"
 
       val result = for {
-        _ <- updateContributionsBannerDeployLog(region, jsonLog.toString)
-        _ <- purgeDeployLogCache(region)
-      } yield bannerRedeploySuccessful(message, region)
+        _ <- updateBannerDeployLog(region, jsonLog.toString, bannerType)
+        _ <- purgeDeployLogCache(region, bannerType)
+      } yield bannerRedeploySuccessful(message, region, bannerType)
 
-      result.recover { case e => redeployFailed(e)}
+      result.recover { case e => redeployFailed(e, bannerType) }
     }
 
   }
 
-  private def updateContributionsBannerDeployLog(region: ReaderRevenueRegion, bannerDeployLogJson: String): Future[Unit] = {
+  private def updateBannerDeployLog(region: ReaderRevenueRegion, bannerDeployLogJson: String, banner: BannerType): Future[Unit] = {
+    log.info(s"updateBannerDeployLog $banner $region")
```

Comment:
Make sure `toString` of those objects are properly implemented or print the relevant fields manually, as discussed yesterday.